ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```
@@ -204,8 +204,110 @@ if [[ -z "${bastion_subnet}" ]] && [[ -z "${bastion_nsg}" ]]; then
     bastion_nsg="${bastion_name}-nsg" bastion_subnet="${bastion_name}Subnet"
     run_command "az network nsg create -g ${bastion_rg} -n ${bastion_nsg}"
     run_command "az network nsg rule create -g ${bastion_rg} --nsg-name '${bastion_nsg}' -n '${bastion_name}-allow' --priority 1000 --access Allow --source-port-ranges '*' --destination-port-ranges ${open_port}"
-    #subnet cidr for int service is xx.xx.99.0/24, it should be a sub rang of the whole VNet cidr, and not conflicts with master subnet and worker subnet
-    bastion_subnet_cidr=$(echo ${AZURE_VNET_ADDRESS_PREFIXES%/*} | awk -F'.' '{print $1"."$2".99.0/24"}')
+
+    # Query VNET address prefix if not already set (standard IPI case)
+    if [[ -z "${AZURE_VNET_ADDRESS_PREFIXES}" ]]; then
+        echo "Querying VNET address prefix from ${bastion_vnet_name}..."
+        AZURE_VNET_ADDRESS_PREFIXES=$(az network vnet show --resource-group ${bastion_rg} --name ${bastion_vnet_name} --query 'addressSpace.addressPrefixes[0]' -o tsv)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)